### PR TITLE
Add basic test for `PrintDocument.Print` using the default print controller

### PR DESCRIPTION
**Pull request by elinor-fung**
We didn't have any coverage for actually using the default print controller. This adds one basic test (conditional based on what is available on the machine)

Scenario from https://github.com/dotnet/runtime/issues/76538.

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]

File: src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs [hunk 1]
```diff
@@ -22,13 +22,13 @@ public class StandardPrintController : PrintController
...
         {
             Debug.Assert(_dc == null && _graphics == null, "PrintController methods called in the wrong order?");
-            Debug.Assert(_modeHandle != null);
 
             base.OnStartPrint(document, e);
             // the win32 methods below SuppressUnmanagedCodeAttributes so assertin on UnmanagedCodePermission is redundant
             if (!document.PrinterSettings.IsValid)
                 throw new InvalidPrinterException(document.PrinterSettings);
 
+            Debug.Assert(_modeHandle != null);
```
Comment (carlossanlop) [inline src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs:31]: For my own education, why is this being moved?

File: src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs [hunk 2]
```diff
@@ -184,6 +185,26 @@ public void EndPrint_SetValue_ReturnsExpected()
...
+            {
+                document.PrinterSettings.PrinterName = PrintToPdfPrinterName;
+                document.PrinterSettings.PrintFileName = GetTestFilePath();
+                document.PrinterSettings.PrintToFile = true;
+                document.EndPrint += endPrintHandler;
+                document.Print();
+                document.EndPrint -= endPrintHandler;
+            }
+
+            // File may not have finished saving to disk when Print returns,
```
Comment (carlossanlop) [inline src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs:203]: Maybe there's a way to wait for a file to finish writing.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs (hunk 1 above)
Comment (elinor-fung) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs:31]: It gets set by the base `PrintController.OnStartPrint`. I can add a comment if that would help.

I'm pretty sure this has been broken in Debug for ~3 years.

File: src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs:203]: Waiting on disk I/O is notoriously unreliable. Testing printing I/O in general is borderline but coupling that with disk I/O seems like the flakiest test possible.

Comment (elinor-fung) [reply src/libraries/System.Drawing.Common/tests/Printing/PrintDocumentTests.cs:203]: I think this is dependent on how the printer handles things, so I didn't see a good way of dealing with it. I didn't want to do some sort of arbitrary timeout/polling and the main goal was to actually call the default printer controller and make sure it didn't error, not necessarily test the 'Microsoft Print to PDF' printer, so I figured making sure we hit `EndPrint` was a reasonable check.

File: src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs (hunk 1 above)
Comment (elinor-fung) [reply src/libraries/System.Drawing.Common/src/System/Drawing/Printing/DefaultPrintController.cs:31]:
```suggestion
            Debug.Assert(_modeHandle != null, "_modeHandle should have been set by PrintController.OnStartPrint");
```

**Review by danmoseley — APPROVED** [review]

**Pull request merged**